api/api.go

```diff
@@ -73,11 +74,22 @@ func (a *API) ListenAndServe(hostAndPort string) error {
 	return http.ListenAndServe(hostAndPort, a.handler)
 }
 
+func timeRequest(ctx context.Context, w http.ResponseWriter, r *http.Request) context.Context {
+	return context.WithValue(ctx, "_netlify_commerce_timing", time.Now())
+}
+
+func logHandler(ctx context.Context, wp mutil.WriterProxy, req *http.Request) {
+	start := ctx.Value("_netlify_commerce_timing").(time.Time)
+	logrus.Infof("method=%s path=%s status=%v duration=%v", req.Method, req.URL.Path, wp.Status(), time.Since(start))
+}
+
 // NewAPI instantiates a new REST API
 func NewAPI(config *conf.Configuration, db *gorm.DB, mailer *mailer.Mailer) *API {
 	api := &API{config: config, db: db, mailer: mailer, httpClient: &http.Client{}}
 	mux := kami.New()
+	mux.LogHandler = logHandler
```

Comment:
To be honest, I really hope you don't plan on submitting that huge change as 1 single PR.